### PR TITLE
Add config files for Fedora 31

### PR DESCRIPTION
**Pull request by skycastlelily2**
This is needed by Fedora 31 and later test.
Fedora-25.ks won't work any more as root login is disable by default from Fedora 31 on

### CODE REVIEW

File: shared/cfg/guest-os/Linux/Fedora/31.cfg [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    unattended_install, svirt_install:
+        kernel = images/f31-${vm_arch_name}/vmlinuz
+        initrd = images/f31-${vm_arch_name}/initrd.img
+        # Unattended-file does not require any changes
+        unattended_file = unattended/Fedora-31.ks
+        unattended_install.url:
+            # Installation works fine with mem=1024 on methods such as cdrom
+            # but fails ("No space left on device") with methods such as url.
+            mem = 2048
+            url = http://dl.fedoraproject.org/pub/fedora/releases/31/Server/${vm_arch_name}/os/
```
Comment (sathnaga) [inline shared/cfg/guest-os/Linux/Fedora/31.cfg]: wrong link instead use this `https://dl.fedoraproject.org/pub/fedora-secondary/releases/31/Server/${vm_arch_name}/os/`

Comment (skycastlelily2) [reply shared/cfg/guest-os/Linux/Fedora/31.cfg]: Hi,thanks a lot for your review,I have updated the commit.

File: shared/cfg/guest-os/Linux/Fedora/31.cfg [hunk 2]
```diff
@@ -0,0 +1,107 @@
+- 31:
+    variants:
+        - aarch64:
+            vm_arch_name = aarch64
+        - ppc64:
+            vm_arch_name = ppc64
```
Comment (ldoktor) [inline shared/cfg/guest-os/Linux/Fedora/31.cfg]: ppc64 is not available for F31...

File: shared/cfg/guest-os/Linux/Fedora/31.cfg [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+                md5sum_1m_cd1 = eebbca4eb53a276d79063fe30ddc6558
+            unattended_install.url:
+                url = http://dl.fedoraproject.org/pub/fedora/linux/releases/31/Server/${vm_arch_name}/os
+                sha1sum_vmlinuz = f1bc444f795b5c3eb7f7cc229bfa5abb
+                sha1sum_initrd = 63c069ce4a4057abc23dcd54897fb03e
+        ppc64:
+            kernel_params = "console=hvc0 serial"
+            boot_path = ppc/ppc64
+            unattended_install.cdrom:
+                md5sum_cd1 = fdbc46e96cd1c210ab62f7a3ad51b938
```
Comment (ldoktor) [inline shared/cfg/guest-os/Linux/Fedora/31.cfg]: Those are copied from F28. I haven't checked them yet but I doubt they stayed the same for the F31... (especially when there is no ppc64 ISO...)

File: shared/cfg/guest-os/Linux/Fedora/31.cfg [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+                md5sum_cd1 = 2694a40333d0f698fa9b4f7faea18ad7
+                md5sum_1m_cd1 = 4b6e4bbfa315283235ea014245b21d1d
+            unattended_install.url:
+                sha1sum_vmlinuz = 956a22a1134972dc08422b13595aac53
+                sha1sum_initrd = 735a5698f95463f4fbf7e1c90a75c1a2
+        x86_64:
+            kernel_params = "console=tty0 console=ttyS0"
+            unattended_install.cdrom:
+                md5sum_cd1 = 18740b445159c54d10bd887650e8d1d7
+                md5sum_1m_cd1 = cf06d7246d515d06ebb73e6e4a2f547f
```
Comment (ldoktor) [inline shared/cfg/guest-os/Linux/Fedora/31.cfg]: Just downloaded this one and it is suppose to be:

```
md5sum_cd1 = af29363e2c8825948515aac36701fbfa
md5sum_1m_cd1 = 8c09623eee12715472242f844cf69e4e
```

please download the rest and update it accordingly.

Comment (ldoktor) [reply shared/cfg/guest-os/Linux/Fedora/31.cfg]: If unsure you can use:

```
for AAA in *.iso; do
echo "$AAA"
md5sum "$AAA"
dd if="$AAA" bs=1M count=1 | md5sum
done
```
which should list md5sum and md5sum_1m for each iso in the current folder.

File: shared/cfg/guest-os/Linux/Fedora/31.cfg (hunk 2 above)
Comment (skycastlelily2) [reply shared/cfg/guest-os/Linux/Fedora/31.cfg]: yeah,since fedora29,sorry for having not cleaned up this file.

File: shared/cfg/guest-os/Linux/Fedora/31.cfg (hunk 4 above)
Comment (skycastlelily2) [reply shared/cfg/guest-os/Linux/Fedora/31.cfg]: Sure,thanks a lot for your review and help me check this.
Btw,should we change sha1sum_vmlinuz(initrd) to md5sum_vmlinuz(initrd),
or update the hashcodes with sha1sum ones,or remain the same?

**Comment by skycastlelily2** [comment]
> selinux - I'd definitely suggest keeping it enforcing
 > removing HWADDR - this should be added as especially multi-nic or hotplug tests can 
> sometimes fail otherwise

 sure,got it.

>packages - yes, adding avocado and removing standard and python makes sense and again >should work on F25 as well (it was available, just not the default and we don't officially support >F25 anymore so I guess it's fine)
>chkconfig vs. systemctl - again, works well on F25
>  PermitRoot - we already talked about this, should work for both
 yes.

 >  net-tools and sg3_utils - sure, makes sense, why not to add it to the %packages section?
As they may be not existed in the main installation repo,so %package are unable to have them installed

File: shared/cfg/guest-os/Linux/Fedora/31.cfg (hunk 4 above)
Comment (skycastlelily2) [reply shared/cfg/guest-os/Linux/Fedora/31.cfg]: Hi Lukas,
I chose the second way and updated the commit and modified the 28.cfg as the links are outdated,please feel free to tell me if they are not wanted.
Btw, I use head -c 1M xxx | md5sum to create the hashcode on the first 1M  before see your comment.

Comment (ldoktor) [reply shared/cfg/guest-os/Linux/Fedora/31.cfg]: no problem, that is equal ;-)

File: shared/cfg/guest-os/Linux/Fedora/31.cfg [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+                md5sum_cd1 = ca302d6a5c21d1bfe41b871b75197767
+                md5sum_1m_cd1 = d459f5d89931a29d199e605be03c84a9
+            unattended_install.url:
+                sha1sum_vmlinuz = b93d1efcafcf29c1673a4ce371a1f8b43941cfeb
+                sha1sum_initrd = 3de45d411df5624b8d8ef21cd0b44419ab59b12f
+        x86_64:
+            kernel_params = "console=tty0 console=ttyS0"
+            unattended_install.cdrom:
+                md5sum_cd1 = af29363e2c8825948515aac36701fbfa
+                md5sum_1m_cd1 = 8c09623eee12715472242f844cf69e4e
```
Comment (ldoktor) [inline shared/cfg/guest-os/Linux/Fedora/31.cfg]: I only checked x86_64 hashes, relying on you (or users) to find out the other...

File: shared/unattended/Fedora-25.ks [hunk 6]
```diff
@@ -1,12 +1,11 @@
-install
 KVM_TEST_MEDIUM
 GRAPHICAL_OR_TEXT
 lang en_US
 keyboard us
 network --bootproto dhcp --hostname atest-guest
 rootpw 123456
 firewall --enabled --ssh
-selinux --enforcing
+selinux --permissive
```
Comment (ldoktor) [inline shared/unattended/Fedora-25.ks]: Do you have a reason to switch to permissive? Unless you have a great reason to use permissive I'd suggest using enforcing to notify potential issues with selinux. We do want to make sure all usual operations work with selinux enabled and our customers are suppose to use selinux in enforcing mode.

File: shared/unattended/Fedora-25.ks [hunk 7]
```diff
@@ -34,15 +33,17 @@ function ECHO { for TTY in `cat /proc/consoles | awk '{print $NF}'`; do source "
 ECHO "OS install is completed"
 grubby --remove-args="rhgb quiet" --update-kernel=$(grubby --default-kernel)
 dhclient
-chkconfig sshd on
 iptables -F
 systemctl mask tmp.mount
-echo 0 > /selinux/enforce
+selinux --enforcing
```
Comment (ldoktor) [inline shared/unattended/Fedora-25.ks]: hmm, why do you disable it for installation and enable here?

File: shared/unattended/Fedora-25.ks [hunk 8]
```diff
@@ -34,15 +33,17 @@ function ECHO { for TTY in `cat /proc/consoles | awk '{print $NF}'`; do source "
...
 sed -i "/^HWADDR/d" /etc/sysconfig/network-scripts/ifcfg-eth0
-# if package groups were missing from main installation repo
+# if packages were missing from main installation repo
 # try again from installed system
 dnf -y groupinstall c-development development-tools
-# include avocado: allows using this machine with remote runner
-dnf -y install python2-avocado
+dnf -y install net-tools sg3_utils python3-avocado
```
Comment (ldoktor) [inline shared/unattended/Fedora-25.ks]: Oh, that is true, I always forget some pkgs are not available directly in installation but they are available post-install...

File: shared/cfg/guest-os/Linux/Fedora/28.cfg [hunk 9]
```diff
@@ -40,7 +40,7 @@
             # but fails ("No space left on device") with methods such as url.
             mem = 2048
             # Unless overridden use secondary url, because it's the most common one
-            url = http://dl.fedoraproject.org/pub/fedora-secondary/releases/28/Server/${vm_arch_name}/os/
+            url = https://archives.fedoraproject.org/pub/archive/fedora-secondary/releases/28/Server/${vm_arch_name}/os/
```
Comment (ldoktor) [inline shared/cfg/guest-os/Linux/Fedora/28.cfg]: this file should be changed in a different commit as it is not related to F31 at all.

hint: If you need to use "and/also/additionally" in main commit message it usually indicates you are doing multiple things and you should perhaps split... ;-)

File: shared/cfg/guest-os/Linux/Fedora/31.cfg [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+            unattended_install.url:
+                url = http://dl.fedoraproject.org/pub/fedora/linux/releases/31/Server/${vm_arch_name}/os
+                sha1sum_vmlinuz = 5b6f6876e1b5bda314f93893271da0d5777b1f3c
+                sha1sum_initrd = 0d7b4bdccb3ca82e0a221104d9f20eda0cc53309
+        # Shared specific setting
+        unattended_install.url:
+            kernel_params += " inst.repo=${url}"
```
Comment (ldoktor) [inline shared/cfg/guest-os/Linux/Fedora/31.cfg]: You removed the warning that this should be removed in the future (note it's already fixed in RHEL but it doesn't seem to be fixed in F31 so it has to be here...)

Comment (ldoktor) [reply shared/cfg/guest-os/Linux/Fedora/31.cfg]: ^^ I mean would you please include the note here so we can remove this hopefully in 32 (or 33?)

**Comment by ldoktor** [comment]
Btw you mentioned avocado guest test so I gave it a try and it'd be nice to modify that test to not to install the python/python2/python-devel/... pkgs when `avocado` command is available... (not saying it should be included here, but it'd make sense to take a look at that as currently it leads to quite a big delay before the test starts even though python3-avocado is already installed)

File: shared/cfg/guest-os/Linux/Fedora/31.cfg (hunk 5 above)
Comment (skycastlelily2) [reply shared/cfg/guest-os/Linux/Fedora/31.cfg]: I've updated all the hashcodes:)

File: shared/unattended/Fedora-25.ks (hunk 6 above)
Comment (skycastlelily2) [reply shared/unattended/Fedora-25.ks]: Really sorry for forgetting to change it to --enforcing here.

File: shared/cfg/guest-os/Linux/Fedora/28.cfg (hunk 9 above)
Comment (skycastlelily2) [reply shared/cfg/guest-os/Linux/Fedora/28.cfg]: Thanks for your useful hint,will keep it on mind in the future.

File: shared/cfg/guest-os/Linux/Fedora/31.cfg (hunk 10 above)
Comment (skycastlelily2) [reply shared/cfg/guest-os/Linux/Fedora/31.cfg]: ^^sure,I didn't know requesting inst.repo is a bug.

**Comment by skycastlelily2** [comment]
> @skycastlelily2 thank you, it is almost there. Would you please split the F28 related changes and address the little issues I mentioned in-code so we can get things going? Right now I'm running the x86 url installation and it seems to be working well (hasn't finished yet).
Nope,I have updated the commit accordingly.

**Comment by skycastlelily2** [comment]
> Btw you mentioned avocado guest test so I gave it a try and it'd be nice to modify that test to not to install the python/python2/python-devel/... pkgs when `avocado` command is available... (not saying it should be included here, but it'd make sense to take a look at that as currently it leads to quite a big delay before the test starts even though python3-avocado is already installed)

Sure, gonna to work on it.Would you please also  review https://github.com/avocado-framework/avocado-vt/pull/2470?thanks a lot^^

File: shared/cfg/guest-os/Linux/Fedora/31.cfg (hunk 10 above)
Comment (ldoktor) [reply shared/cfg/guest-os/Linux/Fedora/31.cfg]: https://bugzilla.redhat.com/show_bug.cgi?id=1649359

**Comment by ldoktor** [comment]
@sathnaga @clebergnu would any of you please do the 2nd review?

File: shared/cfg/guest-os/Linux/Fedora/28.cfg [hunk 11]
```diff
@@ -93,7 +93,7 @@
                 md5sum_cd1 = 18740b445159c54d10bd887650e8d1d7
                 md5sum_1m_cd1 = cf06d7246d515d06ebb73e6e4a2f547f
             unattended_install.url:
-                url = http://dl.fedoraproject.org/pub/fedora/linux/releases/28/Server/${vm_arch_name}/os
+                url = https://archives.fedoraproject.org/pub/archive/fedora/linux/releases/28/Server/${vm_arch_name}/os
                 sha1sum_vmlinuz = 93b2351535ed8cfe9e3440b4bba2402d
```
Comment (sathnaga) [inline shared/cfg/guest-os/Linux/Fedora/28.cfg:97]: I see md5sum values being updated in place of sha1sum for all(archs) vmlinuz and initrd, not the new change, but better to correct now...

E:g:-
```
# md5sum vmlinuz
93b2351535ed8cfe9e3440b4bba2402d  vmlinuz

sha1sum suppose to be
# sha1sum vmlinuz
238e083e114c48200f80d889f7e32eeb2793e02a  vmlinuz
```

Comment (skycastlelily2) [reply shared/cfg/guest-os/Linux/Fedora/28.cfg]: sure, I have  updated the commit

**Comment by ldoktor** [comment]
@sathnaga thank you for spotting the md5/sha1 thing, it should be clear to go now, right?

**Comment by skycastlelily2** [comment]
Hi Satheesh,
Please feel free to tell me if there are more modifications needed  to have this PR get approved.
I will work on it ASAP so that we can close this soon,thanks.

HI Lukas,
Thanks for your approval and help.

**Comment by sathnaga** [comment]
Thanks @skycastlelily2, will merge this...
